gobblin-runtime/src/main/java/gobblin/runtime/Task.java

```diff
@@ -112,9 +120,10 @@ public void run() {
     Closer closer = Closer.create();
     try {
       // Build the extractor for extracting source schema and data records
-      Extractor extractor = closer.register(new ExtractorDecorator(
+      Extractor extractor = closer.register(new InstrumentedExtractorDecorator(this.taskState,
```

Comment:
So we will be collecting metrics no matter the metric reporter is enabled or not?